pkg/core/xds/inspect/config.go

```diff
@@ -32,9 +32,10 @@ type ProxyConfigInspector struct {
 	meshContext            xds_context.MeshContext
 	snapshotGenerator      *v3.TemplateSnapshotGenerator
 	knownInternalAddresses []string
+	dataplaneInsight       *core_mesh.DataplaneInsightResource
 }
 
-func NewProxyConfigInspector(meshContext xds_context.MeshContext, zone string, knownInternalAddresses []string, hooks ...xds_hooks.ResourceSetHook) (*ProxyConfigInspector, error) {
+func NewProxyConfigInspector(meshContext xds_context.MeshContext, dataplaneInsight *core_mesh.DataplaneInsightResource, zone string, knownInternalAddresses []string, hooks ...xds_hooks.ResourceSetHook) (*ProxyConfigInspector, error) {
```

Comment:
we should probably just pass `p.dataplaneInsight.Spec.Metadata if that's all we need.

Reply:
changed